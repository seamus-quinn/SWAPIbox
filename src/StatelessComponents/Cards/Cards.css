.card {
  border-radius: 5px;
  margin: 2px;
  width: 300px;
  height: 300px;
  background-color: #c29991;
  color: cornsilk;

}

.card-header {
  font-family: 'Emilys Candy', cursive;
}

img {
  width: 50px;
  height: 50px;
}