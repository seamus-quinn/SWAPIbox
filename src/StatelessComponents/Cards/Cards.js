import React from 'react';
import './Cards.css'

const PeopleCard = ({ name, homeworld, species, homeworldPop, addToFavorites, display }) => {
  return (
    <div className='card people-card'>
      <h1>{name}</h1>
      <p>Homeworld: {homeworld}</p>
      <p>Species: {species}</p>
      <p>Homeworld Population: {homeworldPop}</p>
      <button onClick={() => addToFavorites(name, display)}>Favorite</button>
    </div>
  )
}

const VehicleCard = ({name, model, vehicleClass, passengers, addToFavorites, display}) => {
  return (
    <div className='card vehicleCard'>
      <h1>{name}</h1>
      <p>Model: {model}</p>
      <p>Class: {vehicleClass}</p>
      <p>Number of Passengers: {passengers}</p>
      <button onClick={() => addToFavorites(name, display)}>Favorite</button>
    </div>
  )
}

const PlanetCard = ({name, terrain, population, climate, addToFavorites, display, residents}) => {
  const people = residents.map(resident => <p>{resident}</p>)
  console.log(people)
  return (
    <div className='card planetCard'>
      <h1>{name}</h1>
      <p>Terrain: {terrain}</p>
      <p>Population: {population}</p>
      <p>Climate: {climate}</p>
      <div>Residents: {people}</div>
      <button onClick={() => addToFavorites(name, display)}>Favorite</button>
    </div>
  )
}

export {
  PeopleCard,
  VehicleCard,
  PlanetCard
} 