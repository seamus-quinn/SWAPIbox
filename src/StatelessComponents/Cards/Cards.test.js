import React from 'react';
import ReactDOM from 'react-dom'
import { shallow, mount } from 'enzyme';
import { PeopleCard, VehicleCard, PlanetCard } from './Cards';

describe('Cards', () => {

  describe('PeopleCard', () => {
    let wrapper;
    let mockProps;

    beforeEach(() => {
      mockProps = {
        person: {
          name: 'Jerry',
          homeworld: 'Mordor',
          species: 'orc',
          homoworldPop: 2
        },
        addToFavorites: jest.fn(),
        display: 'people'
      }
      wrapper = mount(<PeopleCard {...mockProps} />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call addToFavorites with the correct parameters, when button is clicked', () => {
      wrapper.find('button').simulate('click');

      expect(wrapper.prop(addToFavorites)).toHaveBeenCalledWith('Jerry', 'people')
    })
  })

  describe('VehicleCard', () => {
    let wrapper;
    let mockProps;

    beforeEach(() => {
      mockProps = {
        person: {
          name: 'Razor Scooter',
          model: 'S-Class',
          class: 'coupe',
          passsengers: 2
        },
        addToFavorites: jest.fn(),
        display: 'vehicles'
      }
      wrapper = mount(<VehicleCard {...mockProps} />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call addToFavorites with the correct parameters, when button is clicked', () => {

      console.log(wrapper)
      wrapper.find('button').simulate('click');

      expect(wrapper.prop(addToFavorites)).toHaveBeenCalledWith('Razor Scooter', 'vehicles')
    })
  })

  describe('PlanetCard', () => {
    let wrapper;
    let mockProps;

    beforeEach(() => {
      mockProps = {
        person: {
          name: 'Earth',
          terrain: 'fluffy',
          population: 60,
          Climate: 'sometimes too hot',
          Residents: ['Donald Glover', 'Donald Duck', 'Donald Trump']
        },
        addToFavorites: jest.fn(),
        display: 'planets'
      }
      wrapper = mount(<PeopleCard {...mockProps} />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call addToFavorites with the correct parameters, when button is clicked', () => {
      wrapper.find('button').simulate('click');

      expect(wrapper.prop(addToFavorites)).toHaveBeenCalledWith('Earth', 'planets')
    })
  })
})