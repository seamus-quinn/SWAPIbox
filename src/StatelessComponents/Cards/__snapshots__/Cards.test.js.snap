// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards PeopleCard should match the snapshot 1`] = `undefined`;

exports[`Cards PlanetCard should match the snapshot 1`] = `undefined`;

exports[`Cards VehicleCard should match the snapshot 1`] = `undefined`;
